java-checks/src/main/java/org/sonar/java/checks/CollectionImplementationReferencedCheck.java

```diff
@@ -177,4 +330,44 @@ private static String toSimpleName(String fullyQualifiedName) {
   private static boolean isPublic(ModifiersTree modifiers) {
     return ModifiersUtils.hasModifier(modifiers, Modifier.PUBLIC);
   }
+
+  private static <T> Set<T> union(Set<T> a, Set<T> b) {
+    var set = new HashSet<T>();
+    set.addAll(a);
+    set.addAll(b);
+    return set;
+  }
+
+  private static class MethodBodyVisitor extends BaseTreeVisitor {
+
+    private final Map<String, VariableTree> candidateParametersByName = new HashMap<>();
+
+    public final Set<VariableTree> excludeParameters = new HashSet<>();
```

Comment:
How about renaming this field to `excludedParameters` to make it more clear in usage that it is a variable and not a method?

Reply:
Fixed.